src/content/core/okteto-variables.mdx

```diff
@@ -7,36 +7,259 @@ id: okteto-variables
+
+Example:
+
+```yaml
+deploy:
+  commands:
+    - export MY_VAR=hello
+    - echo $MY_VAR # -> ''
+    - echo "MY_VAR=hello" >> $OKTETO_ENV
+    - echo $MY_VAR # -> 'hello'
+```
+
+:::
+
+## Use Cases
+
+### Expand values in the Okteto manifest
+
+Variables can be used to **expand values** in the the **Okteto manifest** (see `${PORT}`)
+
+```yaml
+# okteto.yml
+dev:
+  api:
+    command: npm start
+    environment:
+      PORT: ${PORT}
+```
+
+### Deploy, destroy and test commands
+
+Variables can be used in **deploy**, **destroy** and **test** commands in the Okteto manifest (see `${ENV}`)
+
+```yaml
+# okteto.yml
+deploy:
+  commands:
+    - terraform init -backend-config=./config/${ENV}/backend.hcl
+    - terraform plan -var-file=./config/${ENV}/terraform.tfvars -out tfplan
+    - terraform apply tfplan
+```
+
+
+Additionally, variables are injected as environment variable of the Okteto CLI so your tools or scripts can use them
+
+```yaml
+# Makefile
+.PHONY: deploy
+deploy:
+    terraform init -backend-config=./config/${ENV}/backend.hcl
+    terraform plan -var-file=./config/${ENV}/terraform.tfvars -out tfplan
+    terraform apply tfplan
+```
+```yaml
+# okteto.yml
+deploy:
+  commands:
+    - make deploy
+```
+
+
+
+### Sharing variables between commands with $OKTETO_ENV
+
+As mentioned in the section [Scope](#scope), variables defined in a specific command item are only available to that item by default.
+
+We understand that projects of a certain size and maturity often need to automate such steps, and often that require sharing values between steps. For such scenario we recommend using the special variable called `$OKTETO_ENV`.
+
+```yaml
+test:
+  commands:
+    # Example: some infrastructure has been provisioned
+    - terraform init -backend-config=./config/${ENV}/backend.hcl
+    - terraform plan -var-file=./config/${ENV}/terraform.tfvars -out tfplan
+    - terraform apply tfplan
+    - terraform output -json > tfout.json
+
+    # I make the following var available to other steps
+    - SERVICE_ENDPOINT=$(cat tfout.json | jq -r '.service_endpoint') >> $OKTETO_ENV
+
+    # the following step will have access to SERVICE_ENDPOINT
+    - echo "Running tests against ${SERVICE_ENDPOINT}..."
+```
+
+Another example for the same functionality, can be for example, creating the infrastructure, generating a `.env` file, and then deploy the application.
+
+```yaml
+deploy:
+  commands:
+  - name: Create AWS infrastructure
+    command: |
+      set -e
+      bucketName="${OKTETO_NAMESPACE}-okteto-tacos-shop"
+      terraform apply -input=false -var "bucket_name=$bucketName" -auto-approve
+      echo "S3_BUCKET_NAME=$resourceName" >> "$OKTETO_ENV"
+
+  - name: Create .env
+    command: |
+      set -e
+      echo "S3_BUCKET_NAME=$S3_BUCKET_NAME" > .env
+
+  - name: Deploy the app
+    command: |
+      set -e
+      make deploy
+```
+
+:::note
+
+Here are some highlights, and clarifications on the example above:
```

Comment:
I would also add a sample using a multi-line command. Multiline commands can be very convenient to avoid the usage of `OKTETO_ENV`.

Reply:
Agree, I've added it 👍